packages/admin/src/app/shared/components/table-header/table-header.component.ts

```diff
@@ -19,30 +31,48 @@ interface QueryParams {
   standalone: true,
   imports: [
     FontAwesomeModule,
+    MatBadgeModule,
     MatButtonModule,
     RouterLink,
     MatIconButton,
     MatTooltipModule,
+    OverlayModule,
+    FeatureSearchAndFilterComponent,
+    NgClass,
   ],
   templateUrl: './table-header.component.html',
   styleUrl: './table-header.component.scss',
 })
 export class TableHeaderComponent {
+  filterOpen = false;
+  filterOffsetY = 20;
   private router = inject(Router);
   private activatedRoute = inject(ActivatedRoute);
+  readonly featureSearchAndFilterStore = inject(FeatureSearchAndFilterStore);
+
   tableName = input.required<string>();
   createLinkQueryParams = input<QueryParams>();
   enableFilter = input<boolean>(true);
-  enableSearch = input<boolean>(true);
 
-  // hook these up when we have the functionality
-  search = output<string>();
-  filter = output<string>();
+  onSearchTextChanged = output<string>();
+  onFilterClosed = output();
 
   faCirclePlus = faCirclePlus;
   faBars = faBars;
   faMagnifyingGlass = faMagnifyingGlass;
 
+  readonly navigation = this.router.events.pipe(
+    takeUntilDestroyed(),
+    filter(event => event instanceof NavigationEnd),
+  );
+
+  constructor() {
+    this.navigation.subscribe(event => {
```

Comment:
You should make sure this is unsubscribed.  By adding `takeUntilDestroyed(this.destroyRef)` in your pipe angular will handle it.  See other examples in the code.  Since this is in the constructore phase or class root level, you don't need to pass this.destroyRef to it, so just `takeUntilDestroyed()` at the end of the pipe will work.